main.cpp

```diff
@@ -21,41 +24,72 @@ cstring remove_extension(cstring filename) {
     return filename.substr(0, idx);
 }
 
+const IR::P4Program *prune_statements(const IR::P4Program *program,
```

Comment:
The program has gotten large enough that we should use individual functions. For example, the purpose of this loop is to produce a pruned P4 program.